### ISSUE TITLE
Release v3.0.0 

### ISSUE THREAD

**Issue by kocio-pl** [gravitystorm/openstreetmap-carto#2517]
I'd like to make this new release next week. Any special problems or wishes, like specific PRs to be included?

**Comment by sommerluk** [comment]
Still myself I do not fully really understand the tecnical problem that blocs #2470 . While #2470 itself is not so urgent (not necesarry to have it in 3.0.0 yet), I wonder if the mentioned problems are related to minimum software versions (mapnik-reference?) and to a mayor release of openstreetmap-carto. If so, I think they should be solved before 3.0.0

**Comment by kocio-pl** [comment]
Seems that only @pnorman knows what needs to be done. I would be glad to merge this code before this release, but that depends on fixing the documentation in a week or so. It's OK for me to wait few more days if I know it's needed - that's why I've opened this ticket.

**Comment by pnorman** [comment]
> Seems that only @pnorman knows what needs to be done. I would be glad to merge this code before this release, but that depends on fixing the documentation in a week or so. It's OK for me to wait few more days if I know it's needed - that's why I've opened this ticket.

#2470 is blocked because `carto project.mml` results in an error, and no where do we instruct people to do a different command!

I opened #2521 to make Travis check what we document, but this doesn't unblock #2470, it will make #2470 fail when rebased.

**Comment by kocio-pl** [comment]
What should be done after merging #2521 to make #2470 mergable again?

**Comment by kocio-pl** [comment]
Sorry, wrong button...

**Comment by kocio-pl** [comment]
Looks like default API 2.3.0 in CartoCSS was meant to allow smooth migration path:

https://github.com/mapbox/carto/pull/433#issuecomment-191365800

Maybe now it could be bumped to 3.0.0, but it would be external change and I don't know how likely it is and how would we react to make use of this change. This would probably mean we should at least update INSTALL.md anyway to include Carto >= 0.16.4 (given the default API change was in this version) and also wait for OSM server deploy.

Paul said that he plans to change the documentation and I thought it will just mention to use proper `-a "3.0.0"` option for MML->XML compilation, which would be much easier, does not rely on external change and is possible right now. So I'm confused that now he tries to just update Travis checking, which doesn't fix any problems, just make them explicit.

We need some fix anyway, but I'm not sure any more which path we want to take?

**Comment by pnorman** [comment]
Yes, I'd prefer to change the documentation, and unintentionally gave the impression that I would be doing that change.

The travis change is because the passing state of travis builds was causing people to think that there were no technical issues with the PRs, and it's what I should have done when I had last changed travis

**Comment by kocio-pl** [comment]
But after we change the documentation this new PR will be misleading (people thinking that something is broken with their code), so what's the use of merging it at all?

**Comment by pnorman** [comment]
A documentation update will render the PR moot, but we don't have that yet.

**Comment by pnorman** [comment]
#2470 is unblocked, but isn't a good candidate for merging right before a release. I'd like to see it in master longer so we can shake out any problems.

I'll do up release notes and tag a release.

**Comment by kocio-pl** [comment]
- http://www.openstreetmap.org/user/pnorman/diary/40114
- https://github.com/openstreetmap/chef/issues/105
